v8go.cc

```diff
@@ -1168,14 +1168,24 @@ static void buildCallArguments(Isolate* iso, Local<Value> *argv, int argc, Value
   }
 }
 
-RtnValue FunctionCall(ValuePtr ptr, int argc, ValuePtr args[]) {
+RtnValue FunctionCall(ValuePtr ptr, ValuePtr recv, int argc, ValuePtr args[]) {
   LOCAL_VALUE(ptr)
+
   RtnValue rtn = {nullptr, nullptr};
   Local<Function> fn = Local<Function>::Cast(value);
   Local<Value> argv[argc];
   buildCallArguments(iso, argv, argc, args);
-  Local<Value> recv = Undefined(iso);
-  MaybeLocal<Value> result = fn->Call(local_ctx, recv, argc, argv);
+
+  MaybeLocal<Value> result;
+
+  if (recv == NULL) {
```

Comment:
This feels like an unnecessary conditional, since it is just distinguishing between different call sites.  That means we can remove the conditional by moving code to the caller.

I've opened https://github.com/rogchap/v8go/pull/163 to allow the Go code to explicitly pass an undefined value, so it can be treated like any other value here.